boards/kernel_layout.ld

```diff
@@ -75,6 +75,18 @@ SECTIONS
         KEEP(*(.vectors .vectors.*))
         KEEP(*(.irqs))
 
+        /* RISC-V
+         * There is no vector table in RISCV, so .vectors and .irqs will be
+         * empty. Instead, the _start function needs to be first in the binary
+         * for it to correctly be executed. We also need to include the trap
+         * handler assembly function.
+         *
+         * These are expected to just be empty on other platforms so they
+         * shouldn't have any effect.
+         */
+        KEEP(*(.riscv.start));
+        KEEP(*(.riscv.trap));
```

Comment:
This looks good -- perhaps we should consider renaming the vectors and irqs sections to `arm.*`

Not something that needs fixing in this PR though -- wish GH had a 'this is really a comment not something that needs addressing' option :/

Reply:
Agree with Pat, but more generally does it make sense to separate ARM kernel_layout and RISC-V kernel_layout?